### ISSUE TITLE
Discord/Slack Intergration for channels

### ISSUE THREAD

**Issue by CC52-dev** [zed-industries/zed#14434]
### Check for existing issues

- [X] Completed

### Describe the feature

I would like to request an integration within the channels feature. Me and my team love ot, but since it doesn't sync with slack, it makes it difficult. Discord is also a place with a lot of devs so requesting that as well.

**Comment by CC52-dev** [comment]
Hopefully this will be easily implementation. Also how long on windows support

**Comment by Moshyfawn** [comment]
These platforms have completely separate APIs, so I wouldn't put them together in a single request. I'd advise creating a separate feature request for the other platform, although it might actually be the case of being able to provide the config in userland settings if implemented via #11460 🤔

Related:
- #11460

**Comment by CC52-dev** [comment]
That Is true. Only problem is you would have to do a **lot** Of things and it would be nice to natively have it

**Comment by notpeter** [comment]
I totally understand why this would be a really powerful integration, it's not something we are likely to prioritize building in the near term.

We currently use LiveKit for Audio/Screensharing in collaboration and GitHub for login/identity management.  Practically speaking either of these suggested integrations would require us to not only integrate with Discord/Slack Platform APIs to initiate/manage/control Huddles/Calls, it would also require implementing some method to integrate with their identity/authentication frameworks as well.

Meaningfully though, there would also be a room mapping issue -- in the Slack case I believe it would require a 1:1 mapping between Zed channels and Slack rooms (because otherwise someone else can't join a huddle already in progress).  I imagine you'd have similar issues with Discord.

And implementing just one would benefit a small percentage of users as there is significant fragmentation there (Slack, Discord, Microsoft Teams).

Cool idea though, not impossible, but not something we're likely do anytime soon.
